python/pyiceberg/typedef.py

```diff
@@ -85,16 +86,24 @@ class Record(StructProtocol):
 
     def __init__(self, *data: Union[Any, StructProtocol]) -> None:
         self._data = list(data)
+        for idx, value in enumerate(data):
+            self[idx] = value
 
-    def set(self, pos: int, value: Any) -> None:
+    def __setitem__(self, pos: int, value: Any) -> None:
```

Comment:
I like allowing the use of `__setitem__` and `__getitem__` but do we want to make a breaking change?